src/tools/genpolicy/src/registry.rs

```diff
@@ -133,28 +163,87 @@ impl Container {
         debug!("Getting process field from docker config layer...");
         let docker_config = &self.config_layer.config;
 
+        // Defaults to start with if we can't work out the correct values.
+        process.User.UID = 0;
+        process.User.GID = 0;
+
+        /*
+         * The user field may:
+         *
+         * 1. Be empty
+         * 2. Contain only a UID
+         * 3. Contain a UID:GID pair, in that format
+         * 4. Contain a user name, which we need to translate into a UID/GID pair
```

Comment:
This list is not exhaustive[^1]: we might also get a `username:gid`, `uid:groupname` etc. I think it might be better to split at `:` and then treat the first item as in the `else` block below (213ff.). If there is a second item, we can either leave the int parsing (203ff.) for now , or you could add similar functionality for `/etc/group`.



[^1]: https://github.com/opencontainers/image-spec/blob/8797c3fedb77200c102c816ddcd295bb1e19d3e3/config.md?plain=1#L140

Reply:
@burgerdev @HernanGatta , I think it's OK to file a github issue for future improvements and make those improvements later.